pretix_eth/static/pretix_eth/question-type.js

```diff
@@ -1,7 +1,143 @@
+
+  // Get list of accounts of the connected wallet
+  const accounts = await web3.eth.getAccounts();
+
+  // MetaMask does not give you all accounts, only the selected account
+  selectedAccount = accounts[0];
+
+  document.querySelector("#selected-account").textContent = selectedAccount;
+
+  // Get a handl
+  const template = document.querySelector("#template-balance");
+  const accountContainer = document.querySelector("#accounts");
+
+  // Purge UI elements any previously loaded accounts
+  accountContainer.innerHTML = '';
+
+  // Go through all accounts and get their ETH balance
+  const rowResolvers = accounts.map(async (address) => {
+    const clone = template.content.cloneNode(true);
+    clone.querySelector(".address").textContent = address;
+    accountContainer.appendChild(clone);
+  });
+
+  // Because rendering account does its own RPC commucation
+  // with Ethereum node, we do not want to display any results
+  // until data for all accounts is loaded
+  await Promise.all(rowResolvers);
+
+  // Display fully loaded UI for wallet data
+  document.querySelector("#prepare").style.display = "none";
+  document.querySelector("#connected").style.display = "block";
+}
+
+async function refreshAccountData() {
+  // If any current data is displayed when
+  // the user is switching acounts in the wallet
+  // immediate hide this data
+  document.querySelector("#connected").style.display = "none";
+  document.querySelector("#prepare").style.display = "block";
+
+  // Disable button while UI is loading.
+  // fetchAccountData() will take a while as it communicates
+  // with Ethereum node via JSON-RPC and loads chain data
+  // over an API call.
+  document.querySelector("#btn-connect").setAttribute("disabled", "disabled")
+  await fetchAccountData(provider);
+  document.querySelector("#btn-connect").removeAttribute("disabled")
+}
+
+async function onConnect() {
+
+  // Setting this null forces to show the dialogue every time
+  // regardless if we play around with a cacheProvider settings
+  // in our localhost.
+  web3Modal.providerController.cachedProvider = null;
+  try {
+    provider = await web3Modal.connect();
+  } catch(e) {
+    console.log("Could not get a wallet connection", e);
+    return;
+  }
+  
+  const web3 = new Web3(provider);
+  const accounts = await web3.eth.getAccounts();
+  document.querySelector('#id_3-question_1').value = accounts[0];
+
+  await refreshAccountData();
+}
+
+async function onDisconnect() {
+
+  if(provider.close) {
+    await provider.close();
+    provider = null;
+  }
+
+  selectedAccount = null;
+
+  // Set the UI back to the initial state
+  document.querySelector("#prepare").style.display = "block";
+  document.querySelector("#connected").style.display = "none";
+}
+
+window.addEventListener('load', async () => {
+  init();
+  document.querySelector("#btn-connect").addEventListener("click", onConnect);
+  document.querySelector("#btn-disconnect").addEventListener("click", onDisconnect);
```

Comment:
do we have a disconnect button? Do not see much use in our case for it

Reply:
Agreed, will cut this out.